app/assets/javascripts/auth_source_ldap.js

```diff
@@ -15,4 +15,33 @@ function change_ldap_port(item) {
       }
     }
   }
+}
+function test_connection(test_connection_url){
+    if($("#aid_test_connection_button_auth_source_ldap").prop('disabled') != true ){
+      var serialized_form_data = $("form").serializeObject();
```

Comment:
I'm not much of a JS person I'm afraid, but I wonder what the difference between serializeObject and the usual serialize() is?  (I think we use that in the host_edit.js for submitting that form on AJAX.)